frontend/dockerfile/dockerfile2llb/convert.go

```diff
@@ -269,7 +279,16 @@ func toDispatchState(ctx context.Context, dt []byte, opt ConvertOpt) (*dispatchS
 		}
 
 		if v := st.Platform; v != "" {
-			v, u, err := shlex.ProcessWordWithMatches(v, metaArgsToMap(optMetaArgs))
+			result, err := shlex.ProcessWordWithMatches(v, metaArgsToMap(optMetaArgs))
+			v := result.Word
+			u := result.Matched
+
+			if len(result.Unmatched) > 0 {
```

Comment:
Looks like a candidate for helper func: `reportUnusedArgs(m, opt.Warn, st.Location)`